src/bootstrap.coffee

```diff
@@ -36,25 +36,42 @@ loadPreloadedApps = ->
 		utils.mixpanelTrack('Loading preloaded apps failed', { error: err })
 
 exchangeKey = ->
-	resinApi.get
-		resource: 'device'
-		options:
-			filter:
-				uuid: userConfig.uuid
-		customOptions:
-			apikey: userConfig.apiKey
-	.catchReturn([])
-	.timeout(config.apiTimeout)
-	.then ([ device ]) ->
-		if not device?
-			throw new ExchangeKeyError("Couldn't fetch device with provisioning key")
-		# We found the device, we can try to generate a working device key for it
-		request.postAsync("#{config.apiEndpoint}/api-key/device/#{device.id}/device-key")
-		.spread (res, body) ->
-			if res.status != 200
-				throw new ExchangeKeyError("Couldn't generate device key with provisioning key")
-			userConfig.deviceApiKey = body
-		.return(device)
+	Promise.try ->
+		# If we have an existing device key we first check if it's valid, because if it is we can just use that
+		if userConfig.deviceApiKey?
+			resinApi.get
```

Comment:
You could deduplicate lines 42-51 and 56-64 into a `getDevice(uuid, apikey)` function.